File: Java/ToyLanguageGUI_exam_final/src/domain/statements/PrintStmt.java
package domain.statements;

import domain.PrgState;
import domain.expressions.Expression;
import utils.IDictionary;
import utils.IHeap;
import utils.IOutList;

public class PrintStmt implements Statement {
    private Expression expr;
    public PrintStmt(Expression e)
    {
        expr = e;
    }

    @Override
    public PrgState execute(PrgState p)
    {
        //try {
            IOutList<Integer> list = p.getMessages();
            IDictionary<String, Integer> dict = p.getSymbols();
            IHeap<Integer,Integer> h = p.getHeap();

            int res = expr.eval(dict,h);
            list.addMsg(res);
        //}
        //catch(StackException se){
         //   System.out.println(se.getMessage());}
        return null;
    }

    @Override
    public String toString()
    {
        return "print(" + expr + ")";
    }
}

File: ToyInterpretor/ToyInterpretor/Domain/Statements/PrintStmt.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using ToyInterpretor.Domain.Expressions;
using ToyInterpretor.Utils;

namespace ToyInterpretor.Domain.Statements
{
    class PrintStmt : Statement
    {
        private Expression expr;
        public PrintStmt(Expression e)
        {
            expr = e;
        }

        
        public PrgState execute(PrgState p)
        {
         
           IOutList<int> list = p.getMessages();
           ISymTable<String, int> st = p.getSymbols();
           int res = expr.eval(st);
           list.addMsg(res);
      
            return p;
        }

        
        public override String ToString()
        {
            return "print(" + expr + ")";
        }
    }
}
